### PR TITLE
Improve Help Logs

### LINKED ISSUE #955: An app was not requested, can not continue
see image
enviroment: windows 10 debug x64 with vs 2022
code branch: master

### PR DESCRIPTION
**Pull request by tomadamatkinson**
## Description

Displays help menu when running samples without a target app. This should help users new to the project get used to the command line options

image

Fixes #955

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [ ] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [ ] I have tested the sample on at least one compliant Vulkan implementation
- [ ] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [ ] Any dependent assets have been merged and published in downstream modules
- [ ] For new samples, I have added a paragraph with a summary to the appropriate chapter in the readme of the folder that the sample belongs to [e.g. api samples readme](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/samples/api/README.adoc)
- [ ] For new samples, I have added a tutorial README.md file to guide users through what they need to know to implement code using this feature. For example, see [conditional_rendering](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/samples/extensions/conditional_rendering)
- [ ] For new samples, I have added a link to the [Antora navigation](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/antora/modules/ROOT/nav.adoc) so that the sample will be listed at the Vulkan documentation site

### CODE REVIEW

**Review by gary-sweet — DISMISSED** [review]

File: framework/platform/platform.cpp [hunk 1]
```diff
@@ -116,6 +120,16 @@ ExitCode Platform::initialize(const std::vector<Plugin *> &plugins)
...
 	}
 
+	if (!app_requested())
+	{
+		LOGE("An app was not requested, can not continue");
+		for (auto &help : parser->help())
+		{
+			LOGI(help);
+		}
+		return ExitCode::Help;
```
Comment (asuessenbach) [inline framework/platform/platform.cpp:122]: With this change, the help text is written twice. Once here, and once in `Platform::terminate`.
That is, you should either `LOGI(help)` here or return `ExitCode::Help`, but not both.
Probably the same holds for lines 83-87.

**Review by asuessenbach — COMMENTED** [review]
Maybe one could also slightly modify the help text.
I would not identify "sample_id" or "ID of the sample to run" as the sample name.
Moreover, adding a simple example might be instructive, like
"To run one specific sample call
vulkan_samples sample hello_triangle", or so.

File: framework/platform/platform.cpp (hunk 1 above)
Comment (tomadamatkinson) [reply framework/platform/platform.cpp:122]: Ah nice spot!

**Review by gary-sweet — APPROVED** [review]

**Review by asuessenbach — APPROVED** [review]

**Comment by marty-johnson59** [comment]
Merging w/ 2 approvals as this looks pretty simple:)

**Comment by SaschaWillems** [comment]
I think we should at least include a concrete sample. The list of options is so long, that (at least for me on windows) I need to scroll down to the see the app option, the one that's most important. I think people will still get confused with that large no. of options without an example on how to simply run a sample.

It also acts very odd on Windows...

**Pull request merged**